### ISSUE TITLE
`@stylistic/js/func-call-spacing`: false positive

### ISSUE THREAD

**Issue by Eugeno** [eslint-stylistic/eslint-stylistic#460]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

```
const test = new Map<number, (keyof SomeInterface)[]>([])

// Unexpected whitespace between function name and paren
```

### Reproduction

https://stackblitz.com/edit/github-ua4adn-rr1cii?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
The reason for this issue is that you are using the JS version of this rule to check content with TypeScript types.

We recommend using [`@stylistic/eslint-plugin`](https://eslint.style/packages/default) as it includes the rules for both JavaScript and TypeScript.

Additionally, `func-call-spacing` has been renamed to `function-call-spacing`.
https://eslint.style/rules/js/func-call-spacing#js-func-call-spacing
